kibana/lib/openshift_logging_plugin/public/hack.js

```diff
@@ -0,0 +1,67 @@
+import chrome from 'ui/chrome';
+import { uiModules } from 'ui/modules';
+const app = uiModules.get('apps/openshift_logging_plugin/index');
+
+function redirectOnSessionTimeout($window) {
+    document.cookie = "name=_oauth_proxy; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/;"
+    const APP_ROOT = `${chrome.getBasePath()}`;
+    const path = chrome.removeBasePath($window.location.pathname);
+    window.location.href = APP_ROOT.substring(0,path.length + 1) + '/oauth/sign_out'
+}
+
+app.factory('errorInterceptor', function ($q, $window) {
+
+    return {
+        responseError: function (response) {
+          console.log(response)
+          if (response.status == 401) { 
+              console.log("Session expired. Redirecting to sign_off to refresh")
+              redirectOnSessionTimeout($window);
+          }
+
+          // If unhandled, we just pass the error on to the next handler.
+          return $q.reject(response);
+        }
+    };
+});
+
+/**
+ * Make sure that we add the interceptor to the existing ones.
+ */
+app.config(function($httpProvider) {
+    $httpProvider.interceptors.push('errorInterceptor');
+});
+
+/**
+ * Setup a wrapper around fetch so that we can
+ * handle session timeouts on ajax calls made
+ * by the kfetch component
+ * @param $window
+ */
+function setupResponseErrorHandler($window) {
+    if (!window.fetch) {
```

Comment:
Without being a pro here in AngularJS, what is the difference between using the native `window` object vs. `$window`. Or is this a typo?

Reply:
This is a total cut/paste error from some bits I grabbed from OD.  I'm actually not certain why it doesnt fail here other then maybe they are the same thing from a javascript perspective.  Note i believe that certain objects like `document` and `window` are always available.  I will correct them for consistency

Reply:
Yes, `document` and `window` are always available on the browser. However, since browsers don't provide the full compatibility with all methods for `document` and `window`, JS framework makers always provide wrappers/shims for these objects. If we can you the `$window` from Angular (if it is the same as `window) then we should do so, because you never know what old browser are used out there.

Reply:
further info.. trying to investigate if it effects the behavior i see https://stackoverflow.com/questions/28906180/difference-between-window-and-window-in-ionicframework#:~:text=window%20is%20a%20browser%20object,document.&text=where%20%24window%20is%20a%20service,make%20use%20of%20its%20APIs.